Code/Framework/AzCore/AzCore/DOM/DomPatch.cpp

```diff
@@ -585,6 +585,26 @@ namespace AZ::Dom
         return AZ::Success();
     }
 
+    void CombinePatchOutcomes(PatchOutcome& lhs, const PatchOutcome& rhs)
+    {
+        if (!rhs.IsSuccess())
+        {
+            if (lhs.IsSuccess())
+            {
+                lhs = rhs;
+            }
+            else
+            {
+                AZStd::string newError;
+                newError.reserve(lhs.GetError().length() + rhs.GetError().length() + 1);
+                newError.append(lhs.TakeError());
+                newError.append("\n");
+                newError.append(rhs.GetError());
+                lhs = AZ::Failure<AZStd::string>(AZStd::move(newError));
+            }
+        }
+    }
```

Comment:
Hmm yeah I debated on making this an rvalue but wasn't sure I wanted to demand it - in usage so far it's always been rvalues though, so I'll give that a go (not going to take the suggestion because I have to update the header and it looks like this is mutating `lhs.GetError()` which is an lvalue)

Reply:
Updated to use an rvalue